File: backend/src/main/java/ch/ffhs/search/model/Song.java
package ch.ffhs.search.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.solr.core.mapping.Indexed;
import org.springframework.data.solr.core.mapping.SolrDocument;

@SolrDocument(solrCoreName = "mll")
public class Song {

    @Id
    @Indexed(name = "id", type = "string")
    private String id;

    @Indexed(name = "artist", type = "string")
    private String artist;

    @Indexed(name = "song", type = "string")
    private String song;

    @Indexed(name = "lyrics", type = "string")
    private String lyrics;

    @Indexed(name = "genre", type = "string")
    private String genre;

    @Indexed(name = "language", type = "string")
    private String language;

    public SongDto getDto(){
        SongDto dto = new SongDto();
        dto.id = this.id;
        dto.artist = this.artist;
        dto.song = this.song;
        dto.lyrics = this.lyrics;
        dto.genre = this.genre;
        dto.language = this.language;

        return dto;
    }
}

File: frontend/src/modules/model/song.ts
import {observable} from "mobx";
import {ISongData} from "../../util/apiTypings";


export default class Song{
    @observable id:string = "";
    @observable artist:string = "";
    @observable song:string = "";
    @observable lyrics:string = "";
    @observable genre:string = "";
    @observable language:string = "";

    /*constructor(id:string, artist:string, song:string, lyrics:string, genre:string, language:string){
        this.id = id;
        this.artist = artist;
        this.song = song;
        this.lyrics = lyrics;
        this.genre = genre;
        this.language = language;
    }*/

    public static createFromISongData(dto:ISongData){
        const song = new Song();
        song.id = dto.id;
        song.artist = dto.artist;
        song.song = dto.song;
        song.lyrics = dto.lyrics;
        song.genre = dto.genre;
        song.language = dto.language;

        return song;
    }
}